GUI Undo and Redo

ConorS0505 opened the pull request:
## Proposed Changes
Make undo and redo work in the GUI
  
## Types of Changes
- [ ] Bugfix (Change to a current feature to improve functionality or prevent some issue)
- [x] New feature (Brand new functionality that you want added to the codebase)
- [ ] Other (Anything that doesn't quite fit the above criteria)

ConorS0505 committed:
GUI Undo Redo minus Rename

Undo/Redo actions work for everything except for renaming a function. Loading to a new file clears out the list of mementos, so you cannot undo after loading a file (by design).

ConorS0505 committed:
Undoing/Redoing Renaming

Ye, it works.

SerbianSensation reviewed (changes requested):
Bug: Creating a relationship a->b, hitting undo, then redo, then enabling edit mode and clicking class a crashes the program.
Output from Bug:
at mike.view.GUIView.resetLinesFromConversion(GUIView.java:209)
        at mike.view.GUIView.htmlBoxToEditBox(GUIView.java:183)
        at mike.controller.ClassController$3.mousePressed(ClassController.java:76)

Inconvenient thing: When first adding a relationship, parameter, or field, then saving and exiting edit mode, I have to click undo twice for it to do something.

Other than that, looks really good so far!

ConorS0505 committed:
Bugfixes

Fixed the bugs stefan found. A second copy of the memento is no longer put into the controller when saving an edit box, and the program no longer crashes after undo then redo on a relationship, then opening up an editbox.

SerbianSensation reviewed (commented):
Tested it again and it seems to work great, excellent work! Couldn't find anything wrong.

Now onto tests...

SerbianSensation committed:
Merge branch 'develop' into GUI-UndoRedo2-ElectricBugaloo

Commit:
Adds mockito tests to GUIView

ConorS0505 committed:
Updated test string output

Changed some of the assertions to have more detailed string output if the test failed.

SerbianSensation reviewed (approved):
Really good job getting the GUI tests to ( finally ) work on GitHub! We can finally test that stuff and increase the code coverage haha. Looks good to me!
